packages/shylu/shylu_node/tacho/src/impl/Tacho_SupernodeInfo.hpp

```diff
@@ -118,6 +128,26 @@ template <typename ValueType, typename DeviceType> struct SupernodeInfo {
 
     bool do_not_apply_pivots;
 
+    // for using SpMV
+    rowptr_view rowptrU;
+    colind_view colindU;
+    nzvals_view nzvalsU;
+
+    rowptr_view rowptrL;
+    colind_view colindL;
+    nzvals_view nzvalsL;
+
+    bool spmv_explicit_transpose;
+#if defined(KOKKOS_ENABLE_CUDA)
```

Comment:
Similar to above, 

I'm not sure this is sufficient to ensure cusparse is available? Would be helpful to add more conditions to the check though I'm not sure what is sufficient (cusparse might be available by enabling as a tpl, or through kokkos-kernels or both); would something like:

```c++
#if defined(KOKKOS_ENABLE_CUDA) && (defined(TPL_ENABLE_CUSPARSE) || defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE))
```
cover either case?